### PR TITLE
vnc_test: avoid usage of complex tests.NewRandomVMI

### PR DESCRIPTION
**Pull request by dankenigsberg**
`libvmi.New(libvmi.WithResourceMemory("1Mi"))` is simpler and smaller, yet just as good for tests that do not need a guest at all.

```release-note
NONE
```

/sig code-quality

### CODE REVIEW

**Comment by dankenigsberg** [comment]
> There's a good chance we would need to set their equivalent's via libvmi to keep these tests honest

You are right that my first attempt was too naive, but these details are exactly what I'd like to avoid. The tests in this file do not care about networking or Arm. They just to need the VM to start and to have a working VNC.

**Comment by akalenyu** [comment]
> > There's a good chance we would need to set their equivalent's via libvmi to keep these tests honest
> 
> You are right that my first attempt was too naive, but these details are exactly what I'd like to avoid. The tests in this file do not care about networking or Arm. They just to need the VM to start and to have a working VNC.

Definitely

Another pain point here is that `NewRandomVMI` gives out a good amount of memory
according to cirros' requirements, but it does not use a cirros image?

**Comment by dankenigsberg** [comment]
> Another pain point here is that `NewRandomVMI` gives out a good amount of memory
> according to cirros' requirements, but it does not use a cirros image?

Correct. It's not the best function - it make (wrong) assumptions about its users. This PR reduces
```
$ git grep NewRandomVMI\\b | wc -l
33
```
to `32`. I'd like to see this down to zero.

**Comment by dankenigsberg** [comment]
This PR is finally ready for review, sorry for the noise. Failing test is unrelated
```
{ Failure tests/vmi_kernel_boot_test.go:140
Timed out after 60.000s.
VMI Should be successfully deleted
The matcher passed to Eventually returned the following error:
    <*errors.errorString | 0xc004e70060>: 
    Expected an error, got nil
    {
        s: "Expected an error, got nil",
    }
tests/vmi_kernel_boot_test.go:166}
```

**Review by akalenyu — COMMENTED** [review]
:+1:

**Review by akalenyu — APPROVED** [review]

File: tests/vnc_test.go [hunk 1]
```diff
@@ -67,9 +68,10 @@ var _ = Describe("[rfe_id:127][crit:medium][arm64][vendor:cnv-qe@redhat.com][lev
 		BeforeEach(func() {
 			virtClient = kubevirt.Client()
 
-			vmi = tests.NewRandomVMI()
-			Expect(virtClient.RestClient().Post().Resource("virtualmachineinstances").Namespace(testsuite.GetTestNamespace(vmi)).Body(vmi).Do(context.Background()).Error()).To(Succeed())
-			libwait.WaitForSuccessfulVMIStart(vmi)
+			vmi = libvmi.New(libvmi.WithResourceMemory("1Mi"))
+			vmi, err = virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vmi)
```
Comment (AlonaKaplan) [inline tests/vnc_test.go:72]: If you could use the opportunity to get rid of `virtClient` and call `kubevirt.Client()` directly it would be great.

Comment (AlonaKaplan) [inline tests/vnc_test.go:72]: Please mention this change (from `virtClient.RestClient().Post()`) in the commit message.

**Comment by dankenigsberg** [comment]
@akalenyu would you renew your review? I believe I've add the addressed @AlonaKaplan 's comments.

**Review by akalenyu — APPROVED** [review]

**Comment by akalenyu** [comment]
/test pull-kubevirt-e2e-k8s-1.29-sig-storage
likely just an occurrence of https://issues.redhat.com/browse/CNV-36682
@haircommander would it be reasonable to ask for cri-o patch releases that contain https://github.com/cri-o/cri-o/pull/7735?

/cc @tiraboschi @xpivarc @acardace

**Comment by haircommander** [comment]
yup I've started the release process!

File: tests/vnc_test.go [hunk 2]
```diff
@@ -61,15 +61,12 @@ import (
 var _ = Describe("[rfe_id:127][crit:medium][arm64][vendor:cnv-qe@redhat.com][level:component][sig-compute]VNC", decorators.SigCompute, func() {
 
 	var err error
-	var virtClient kubecli.KubevirtClient
```
Comment (xpivarc) [inline tests/vnc_test.go:63]: Wish we could also remove the `err`

Comment (dankenigsberg) [reply tests/vnc_test.go:63]: Oh dear. `err` is mentioned in a maze of nested functions, some of them masking the original one with `:=`. With your permission I would consider it in a follow-up https://github.com/kubevirt/kubevirt/pull/11340 .

**Pull request merged**